doc/api/diagnostics_channel.md

```diff
@@ -407,6 +457,526 @@ channel.subscribe(onMessage);
+* `ctx` {Object} Shared context object to correlate trace events through
+* `thisArg` {any} The receiver to be used for the function call
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {any} The return value of the given function
+
+Trace a synchronous function call. This will always produce `start` and `end`
+events around the execution and may also produce an `error` event if the given
+function throws an error. This will run the given function using
+[`channel.runStores(data, ...)`][] on the `start` channel.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceSync(() => {
+  // Do something
+}, {
+  context: 'something',
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceSync(() => {
+  // Do something
+}, {
+  context: 'something',
+});
+```
+
+#### `tracingChannel.tracePromise(fn[, ctx[, thisArg, [, ...args]]])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `fn` {Function} Promise-returning function to wrap a trace around
+* `ctx` {Object} Shared context object to correlate trace events through
+* `thisArg` {any} The receiver to be used for the function call
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {Promise} Promise returned by the given function
+
+Trace a promise-returning function call. This will always produce `start`,
+`end`, `asyncStart`, and `asyncEnd` events around the execution and may also
+produce an `error` event if the given function throws an error or the returned
+promise rejects. This will run the given function using
+[`channel.runStores(data, ...)`][] on the `start` channel.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.tracePromise(async () => {
+  // Do something
+}, {
+  context: 'something',
+});
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.tracePromise(async () => {
+  // Do something
+}, {
+  context: 'something',
+});
+```
+
+#### `tracingChannel.traceCallback(fn[, position[, ctx[, thisArg, [, ...args]]]])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `fn` {Function} Promise-returning function to wrap a trace around
+* `position` {number} Zero-indexed argument position of expected callback
+* `ctx` {Object} Shared context object to correlate trace events through
+* `thisArg` {any} The receiver to be used for the function call
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {Promise} Promise returned by the given function
+
+Trace a promise-returning function call. This will always produce `start`,
```

Comment:
Is the `fn` expected to return a promise? I suppose the `fn` is a void-returning callback instead.

Reply:
Oh, yep, good catch. Copy/pasted from the `tracePromise` docs and forgot to update that bit. 😅